core/build.gradle

```diff
@@ -8,8 +8,13 @@ plugins {
     id "org.jetbrains.dokka"
 }
 
-java.sourceCompatibility = "$jvm_version"
-java.targetCompatibility = "$jvm_version"
+java {
+    sourceCompatibility = "$jvm_version"
+    targetCompatibility = "$jvm_version"
+    registerFeature('cli') {
```

Comment:
I have never seen this. What is this used for?

Reply:
To have the clikt library as an optional dependency in the generated POM. So that consumers of Kolasu who don't use it don't get the dependency.

Reply:
How does one specify that he wants it when specifying Kolasu as a dependency?

Reply:
I gather that to use it from Gradle you'd have to write something like
```
implementation("com.strumenta.kolasu:kolasu-core:1.x.y")
implementation("com.strumenta.kolasu:kolasu-core:1.x.y") {
        capabilities {
           requireCapability("com.strumenta.kolasu:kolasu-core-cli")
        }
    }
```
This explains it https://blog.gradle.org/optional-dependencies
If it's too complicated we could either make the dependency mandatory (consumers who don't want it can still exclude it) or make it the equivalent of Maven's "provided" which I don't know in Gradle (i.e. consumers who want it must know about it & add the specific dependencies to their project autonomously).